pkg/preflight/collect.go

```diff
@@ -374,15 +375,16 @@ func ensureCollectorInList(list []*troubleshootv1beta2.Collect, collector troubl
 	return append(list, &collector)
 }
 
-func ensureHostCollectorInList(list []*troubleshootv1beta2.HostCollect, collector troubleshootv1beta2.HostCollect) []*troubleshootv1beta2.HostCollect {
-	for _, inList := range list {
-		if collector.CPU != nil && inList.CPU != nil {
-			return list
-		}
-		if collector.Memory != nil && inList.Memory != nil {
-			return list
+// collect ClusterResources earliest in the list so the pod list does not include pods started by collectors
```

Comment:
would have preferred to add this function only once, in pkg/supportbundle, but till we combine the collect for both supportbundle and preflight, that refactor is beyond the scope of this change, so it's duplicated here (several other parts already are)

Reply:
Any issue with putting it in `pkg/collect` and then making it public?

Reply:
thanks, needed a push there :)